### PR TITLE
Notification Mitigation - AppNotificationManager::IsSupported

### PR DESCRIPTION
**Pull request by pmpurifoy**
Branch was based on stable 1.1 branch which led to unnecessary files being merged to main. Previous PR was closed here: https://github.com/microsoft/WindowsAppSDK/pull/2505

Issue:
In elevated scenarios, PushNotifications do not work for both unpackaged + packaged applications and the IsSupported API doesn't include this limitation. Also for AppNotifications, there is no IsSupported API. Elevated scenarios are currently not supported for AppNotifications so we are added an IsElevated check to this newly added IsSupported API.

Fix:
Added IsElevated check to the Push/AppNotification IsSupported API.

Verified:
Changes were tested by manually running scenarios of unpackaged/packaged + non-elevated/elevated and confirming correct behavior of Push/AppNotification APIs.

### CODE REVIEW

File: dev/AppNotifications/AppNotificationManager.cpp [hunk 1]
```diff
@@ -93,8 +93,19 @@ namespace winrt::Microsoft::Windows::AppNotifications::implementation
         return m_activatedEventArgs;
     }
 
+    bool AppNotificationManager::IsSupported()
+    {
+        static bool isSupported{ !PushNotificationHelpers::IsElevated() };
```
[the remark this answers is not in the document]
Comment (pmpurifoy) [reply dev/AppNotifications/AppNotificationManager.cpp:98]: IsSupported in PushNotification vs AppNotifications have similar functions. They both block cloud notifications from their component, but AppNotificationManager allows local toasts which is why it looks a little different/less restrictive than the PushNotificationManager.

File: dev/PushNotifications/PushNotificationUtility.h [hunk 2]
```diff
@@ -138,6 +138,18 @@ namespace winrt::Microsoft::Windows::PushNotifications::Helpers
...
+    inline bool IsElevated()
+    {
+        wil::unique_handle processToken;
+        THROW_IF_WIN32_BOOL_FALSE(OpenProcessToken(GetCurrentProcess(), MAXIMUM_ALLOWED, &processToken));
+
+        auto tokenLabel{ wil::get_token_information<TOKEN_MANDATORY_LABEL>(processToken.get()) };
+        DWORD subAuthority = static_cast<DWORD>(*GetSidSubAuthorityCount(tokenLabel->Label.Sid) - 1);
+
+        // HIGH_RID means the process is elevated
+        return *GetSidSubAuthority(tokenLabel->Label.Sid, subAuthority) == SECURITY_MANDATORY_HIGH_RID;
```
[the remark this answers is not in the document]
Comment (pmpurifoy) [reply dev/PushNotifications/PushNotificationUtility.h]: Nope, that one is much simpler. Replacing.

File: dev/AppNotifications/AppNotificationManager.cpp [hunk 3]
```diff
@@ -406,11 +447,21 @@ namespace winrt::Microsoft::Windows::AppNotifications::implementation
 
     winrt::Windows::Foundation::IAsyncOperation<winrt::Microsoft::Windows::AppNotifications::AppNotificationProgressResult> AppNotificationManager::UpdateAsync(winrt::Microsoft::Windows::AppNotifications::AppNotificationProgressData const data, hstring const tag)
     {
+        if (!IsSupported())
```
Comment (hulumane) [inline dev/AppNotifications/AppNotificationManager.cpp:450]: Is this needed here? I believe the API call below already gates. So ....

File: dev/AppNotifications/AppNotificationManager.cpp [hunk 4]
```diff
@@ -551,6 +627,11 @@ namespace winrt::Microsoft::Windows::AppNotifications::implementation
 
     winrt::Windows::Foundation::IAsyncOperation<winrt::Windows::Foundation::Collections::IVector<winrt::Microsoft::Windows::AppNotifications::AppNotification>> AppNotificationManager::GetAllAsync()
     {
+        if (!IsSupported())
+        {
+            co_return nullptr;
```
Comment (hulumane) [inline dev/AppNotifications/AppNotificationManager.cpp:632]: MAy want to return an empty vecotr here. Not null

File: dev/AppNotifications/AppNotificationManager.cpp (hunk 3 above)
Comment (pmpurifoy) [reply dev/AppNotifications/AppNotificationManager.cpp:450]: This check was just to be thorough, but it is redundant. Will fix next iteration.

File: dev/PushNotifications/PushNotificationUtility.h [hunk 5]
```diff
@@ -138,6 +138,15 @@ namespace winrt::Microsoft::Windows::PushNotifications::Helpers
         return hasSetTaskEntrypoint;
     }
 
+    inline bool IsElevated()
```
Comment (hulumane) [inline dev/PushNotifications/PushNotificationUtility.h:141]: Was this copied over from somewhere else in the SDK? If so, might make sense to move this to the common package.

File: test/TestApps/ToastNotificationsDemoApp/main.cpp [hunk 6]
```diff
@@ -208,9 +209,6 @@ int main()
         std::wcout << std::endl;
     }
 
-    std::wcout << L"Requesting PushNotificationChannel...\n\n";
```
Comment (hulumane) [inline test/TestApps/ToastNotificationsDemoApp/main.cpp:211]: why is this here? Just an orphan we forgot to delete? :(

File: dev/PushNotifications/PushNotificationUtility.h (hunk 5 above)
Comment (pmpurifoy) [reply dev/PushNotifications/PushNotificationUtility.h]: Did not know Security::IntegrityLevel::IsElevated was provided by Security.IntegrityLevel.h. Switching to that helper API instead.

**Review by hulumane — APPROVED** [review]

**Review by BenJKuhn — APPROVED** [review]
